### PR TITLE
OP-1100 Fix API current user variable in MDC

### PR DESCRIPTION
**Pull request by mwithi**
See OP-1100.

### CODE REVIEW

**Review by dbmalkovsky — COMMENTED** [review]
I'm not sure exactly what is in `getPrinipal()` that is why I provided an alternative.

**Comment by mwithi** [comment]
> I'm not sure exactly what is in `getPrinipal()` that is why I provided an alternative.

Principal is the way is called the logged user itself in Spring Security, anyway also your alternative is ok.

File: src/main/java/org/isf/login/rest/LoginController.java [hunk 1]
```diff
@@ -66,17 +67,18 @@ public class LoginController {
...
-	public ResponseEntity<?> authenticateUser(@Valid @RequestBody LoginRequest loginRequest) throws OHAPIException {
+	public ResponseEntity< ? > authenticateUser(@Valid @RequestBody LoginRequest loginRequest) throws OHAPIException {
 		Authentication authentication = authenticationManager.authenticate(
-				new UsernamePasswordAuthenticationToken(loginRequest.getUsername(), loginRequest.getPassword()));
+						new UsernamePasswordAuthenticationToken(loginRequest.getUsername(), loginRequest.getPassword()));
 		SecurityContextHolder.getContext().setAuthentication(authentication);
 		String jwt = tokenProvider.generateJwtToken(authentication, true);
 
 		String userDetails = (String) authentication.getPrincipal();
+		MDC.put("OHUser",loginRequest.getUserName());
```
Comment (mwithi) [inline src/main/java/org/isf/login/rest/LoginController.java:77]:
```suggestion
		MDC.put("OHUser", loginRequest.getUserName());
```

**Review by dbmalkovsky — APPROVED** [review]
LGTM

File: src/main/java/org/isf/login/rest/LoginController.java [hunk 2]
```diff
@@ -66,17 +67,18 @@ public class LoginController {
...
-	public ResponseEntity<?> authenticateUser(@Valid @RequestBody LoginRequest loginRequest) throws OHAPIException {
+	public ResponseEntity< ? > authenticateUser(@Valid @RequestBody LoginRequest loginRequest) throws OHAPIException {
 		Authentication authentication = authenticationManager.authenticate(
-				new UsernamePasswordAuthenticationToken(loginRequest.getUsername(), loginRequest.getPassword()));
+						new UsernamePasswordAuthenticationToken(loginRequest.getUsername(), loginRequest.getPassword()));
 		SecurityContextHolder.getContext().setAuthentication(authentication);
 		String jwt = tokenProvider.generateJwtToken(authentication, true);
 
 		String userDetails = (String) authentication.getPrincipal();
+		MDC.put("OHUser", loginRequest.getUserName());
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/login/rest/LoginController.java:77]:
```suggestion
		MDC.put("OHUser", loginRequest.getUsername());
```

Comment (dbmalkovsky) [reply src/main/java/org/isf/login/rest/LoginController.java:77]: Fix poorly named method to fix compile error.

**Pull request merged**